pkg/crc/machine/certs-expiration.go

```diff
@@ -0,0 +1,67 @@
+package machine
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/code-ready/crc/pkg/crc/cluster"
+	"github.com/code-ready/crc/pkg/crc/machine/bundle"
+	crcssh "github.com/code-ready/crc/pkg/crc/ssh"
+)
+
+/* Returned when a certificate is past its expiration date */
+type certsExpiredError struct {
+	expirationTime time.Time
+}
+
+func (err certsExpiredError) Error() string {
+	return fmt.Sprintf("Instance certificates have expired, they were valid until %s", err.expirationTime.Format(time.RFC822))
+}
+
+type CertExpirationState int
+
+const (
+	Unknown CertExpirationState = iota
+	CertNotExpired
+	CertExpiringIn7Days
+	CertExpired
+)
+
+func (state CertExpirationState) String() string {
+	return [...]string{"Unknown state", "Not Expired", "Expiring in 7 days", "Expired"}[state]
+}
+
+func daysTillExpiration(expirationTime time.Time) int {
+	durationTillExpiration := time.Until(expirationTime)
+	return int(durationTillExpiration.Hours() / 24)
+}
+
+func checkCertsExpiration(expirationTime time.Time) (CertExpirationState, error) {
+	daysLeft := daysTillExpiration(expirationTime)
+	if daysLeft < 0 {
+		return CertExpired, certsExpiredError{expirationTime}
+	} else if daysLeft < 7 {
+		return CertExpiringIn7Days, nil
```

Comment:
@cfergeau This will always give warning that cert is going to expire in 7 days even only 1-2 days left which is wrong and user thinks he still have 7 days to delete/recreate the environment. It should be function instead part of const which take the number of day left as arg.

Reply:
Ah right, maybe it's enough to change the warning from "will expire in 7 days" to "will expire in less than 7 days"?

Reply:
if no `-days` option is possible, say `less than 7 days left`